ocaml/typing/typecore.ml

```diff
@@ -7425,6 +7437,55 @@ and type_expect_jane_syntax
       type_unboxed_constant
         ~loc ~env ~expected_mode ~ty_expected ~explanation ~attributes x
 
+and type_local_expr
```

Comment:
I tried to review this function to confirm that nothing changed from the old world, but I think this would benefit from Zesen's eyes. (And also to preview for him what changes are coming in local land.) @riaqn would you mind taking a look at this `type_local_expr` function to give your read on whether it does the same thing as the old typing code for local/exclave?

Reply:
The case for `Lexp_local` and `Lexp_exclave` looks identitcal to the original code, so it should be fine. I'm not sure about `Lexp_constrain_local` - it's a no-op?

Reply:
That's right -- it's artificial, and only used for `extract_constraint_alloc_mode`.  It'll be going away with n-ary arrows.  Thanks!